peps/pep-9999.rst

```diff
@@ -0,0 +1,616 @@
+- `Wheels over
+  sdists <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L504>`__
+- `Maximize
+  version <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L544>`__
+  (including `Local version
+  tags <https://peps.python.org/pep-0440/#local-version-identifiers>`__,
+  `restricted to locally patched private
+  versions <https://discuss.python.org/t/lets-permit-local-version-label-in-version-specifiers/22781/2>`__)
+- `Maximize specificity of platform
+  tag <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L520>`__
+- `Build
+  tags <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/index/package_finder.py#L535>`__
+  with the first value being an integer interpreted as ascending values
+  being “better”, and the second being a string with lexical sorting
+
+This behavior is characterized by the installer always getting the
+latest version of a package, regardless of the repository that it comes
+from. It is especially useful when all configured repositories are
+equally trusted and well-behaved regarding the distribution
+interchangeability assumption. That interchangeability assumption is
+what makes comparing distributions of a given package meaningful -
+without it, the installer is no longer comparing “apples to apples.” In
+practice, it is common for different indexes to have files that have
+different contents than other indexes, such as builds for special
+hardware, and this version priority behavior can lead to undesirable,
+unexpected outcomes.
+
+Uv refers to `its implementation of the version priority
+behavior <https://docs.astral.sh/uv/pip/compatibility/#packages-that-exist-on-multiple-indexes>`__
+as “``unsafe-best-match``.” Naming is really hard here. On one hand, it
+isn’t accurate to call pip’s default behavior intrinsically “unsafe.”
+The addition of possibly malicious indexes/repositories is what
+introduces concern with this behavior. PEP 708 added a way to restrict
+installers from drawing packages from unexpected, potentially insecure
+indexes. On the other hand, the term “best-match” is technically
+correct, but also misleading. The “best match” varies by user and by
+application. “Best” is technically correct in the sense that it is a
+global optimum according to the match criteria specified above, but that
+is not necessarily what is “best” in a user’s eyes. “Version priority”
+is a proposed term that avoids the concerns with the uv terminology,
+while approximating the behavior in the most user-identifiable way that
+packages are compared.
+
+“Index priority”
+----------------
+
+Relative to “version priority” behavior, this strategy does not collate
```

Comment:
What about adding a leading summary which describes index priority without being relative to version priority?

Reply:
Added. Please see what you think.